src/xb-opcode.c

```diff
@@ -176,9 +214,7 @@ xb_opcode_cmp_val (XbOpcode *self)
 inline gboolean
 xb_opcode_cmp_str (XbOpcode *self)
 {
-	return self->kind == XB_OPCODE_KIND_TEXT ||
-		self->kind == XB_OPCODE_KIND_BOUND_TEXT ||
-		self->kind == XB_OPCODE_KIND_INDEXED_TEXT;
+	return (self->kind & XB_OPCODE_FLAG_TEXT) > 0;
```

Comment:
Couldn’t this be a call to `xb_opcode_has_flag()`?

Reply:
It can indeed, fixed.